src/main/java/cms/gov/madie/measure/services/TestCaseService.java

```diff
@@ -0,0 +1,46 @@
+package cms.gov.madie.measure.services;
+
+import cms.gov.madie.measure.exceptions.ResourceNotFoundException;
+import cms.gov.madie.measure.models.Measure;
+import cms.gov.madie.measure.models.TestCase;
+import cms.gov.madie.measure.repositories.MeasureRepository;
+import org.bson.types.ObjectId;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class TestCaseService {
+
+  private final MeasureRepository measureRepository;
+
+  public TestCaseService(MeasureRepository measureRepository) {
+    this.measureRepository = measureRepository;
+  }
+
+  public TestCase persistTestCase(TestCase testCase, String measureId) {
+    Measure measure = findMeasureById(measureId);
+
+    // mongo doesn't create object id for embedded objects, setting manually
+    testCase.setId(ObjectId.get().toString());
+    if (measure.getTestCases() == null) {
+      measure.setTestCases(List.of(testCase));
+    } else {
+      measure.getTestCases().add(testCase);
+    }
+    measureRepository.save(measure);
+    return testCase;
+  }
+
+  public List<TestCase> findTestCasesByMeasureId(String measureId) {
+    return findMeasureById(measureId).getTestCases();
+  }
+
+  public Measure findMeasureById(String measureId) {
+    Measure measure = measureRepository.findById(measureId).orElse(null);
+    if (measure == null) {
+      throw new ResourceNotFoundException("Measure", measureId);
```

Comment:
I think the way this is written right now would return a 500 to the caller, which probably isn't what we want. In the past I have done a few different things with this kind of check:
- handle this check at the controller level, and throw a ResponseStatusException (provided by Spring). This is very configurable per instance and the error messages returned would look just like any other errors caught by Spring.
- Use @ControllerAdvice to create a global exception handler and map different internal application errors to different response codes / objects
- Handle this check at the controller (or catch this exception) and return a ResponseEntity.notFound() (or similar)

Unless calling the API with an unknown measure ID is truly an internal exception, I think we should handle that and return some nicer response code and message.

Reply:
I thought we have ControllerAdvice that catches the exception and prepares the error response. I'll double-check with frontend changes once I pull your security branch.

Reply:
@nmorasb, there was no exception handler. I've added an exception handler in controller advice with BAD_REQUEST status code. pls, let me know if this works

Reply:
@adongare How about http status NOT_FOUND instead? It informs the caller that their request is acceptable, but the requested resource isn't available at the path.

Reply:
@jkotanchik-SB can do that. was not sure if NOT_FOUND or BAD_REQUEST. Thanks!

Reply:
done